docs/gguf.md

```diff
@@ -0,0 +1,443 @@
+- `mpt.embedding_length`
+- `mpt.layer_count`
+- `mpt.attention.head_count`
+- `mpt.attention.alibi_bias_max`
+- `mpt.attention.clip_kqv`
+
+##### GPT-NeoX
+
+- `gptneox.context_length`
+- `gptneox.embedding_length`
+- `gptneox.layer_count`
+- `gptneox.use_parallel_residual`
+- `gptneox.rope.dimension_count`
+- `gptneox.attention.head_count`
+
+###### Optional
+
+- `gptneox.rope.scale`
+
+##### GPT-J
+
+- `gptj.context_length`
+- `gptj.embedding_length`
+- `gptj.layer_count`
+- `gptj.rope.dimension_count`
+- `gptj.attention.head_count`
+
+###### Optional
+
+- `gptj.rope.scale`
+
+##### GPT-2
+
+- `gpt2.context_length`
+- `gpt2.embedding_length`
+- `gpt2.layer_count`
+- `gpt2.attention.head_count`
+
+##### BLOOM
+
+- `bloom.context_length`
+- `bloom.embedding_length`
+- `bloom.layer_count`
+- `bloom.feedforward_length`
+- `bloom.attention.head_count`
+
+##### Falcon
+
+- `falcon.context_length`
+- `falcon.embedding_length`
+- `falcon.layer_count`
+- `falcon.attention.head_count`
+- `falcon.attention.head_count_kv`
+- `falcon.attention.use_norm`
+
+##### RWKV
+
+**TODO**.
+
+##### Whisper
+
+Keys that do not have types defined should be assumed to share definitions with `llm.` keys.
+(For example, `whisper.context_length` is equivalent to `llm.context_length`.)
+This is because they are both transformer models.
+
+- `whisper.encoder.context_length`
+- `whisper.encoder.embedding_length`
+- `whisper.encoder.layer_count`
+- `whisper.encoder.mels_count: u32`
+- `whisper.encoder.attention.head_count`
+
+- `whisper.decoder.context_length`
+- `whisper.decoder.embedding_length`
+- `whisper.decoder.layer_count`
+- `whisper.decoder.attention.head_count`
+
+#### Prompting
+
+**TODO**: Include prompt format, and/or metadata about how it should be used (instruction, conversation, autocomplete, etc).
+
+### LoRA
+
+**TODO**: Figure out what metadata is needed for LoRA. Probably desired features:
+
+- match an existing model exactly, so that it can't be misapplied
+- be marked as a LoRA so executors won't try to run it by itself
+
+Should this be an architecture, or should it share the details of the original model with additional fields to mark it as a LoRA?
+
+### Tokenizer
+
+The following keys are used to describe the tokenizer of the model. It is recommended that model authors support as many of these as possible, as it will allow for better tokenization quality with supported executors.
+
+#### GGML
+
+GGML supports an embedded vocabulary that may be lossily compressed from a more complete tokenizer. It is simplistic and specific to GGML. This should enable inferencing of the model, but it may not fully capture the nuances of tokenization. When a more accurate tokenizer is available and supported, it should be used instead.
+
+It is not guaranteed to be standardized across models, and may change in the future. It is recommended that model authors use a more standardized tokenizer if possible.
+
+- `tokenizer.ggml.tokens: array[string]`: A list of tokens.
```

Comment:
Add better description:

- `tokenizer.ggml.tokens: array[string]` A list of tokens indexed by the token id used by the model.